src/main/java/crawlercommons/domains/EffectiveTldFinder.java

```diff
@@ -65,26 +96,73 @@
      * A singleton
      */
     private EffectiveTldFinder() {
-        initialize(null);
+        initialize(this.getClass().getResourceAsStream(ETLD_DATA));
     }
 
+    /**
+     * Get singleton instance of EffectiveTldFinder with default configuration.
+     * 
+     * @return singleton instance of EffectiveTldFinder
+     */
     public static EffectiveTldFinder getInstance() {
         if (null == instance) {
             instance = new EffectiveTldFinder();
         }
         return instance;
     }
 
+    /**
+     * (Re)initialize EffectiveTldFinder with built-in public suffix list.
+     * 
+     * @param excludePrivateDomains
+     *            whether to exclude the public suffixes listed in the PRIVATE
+     *            domain section (opposed to &quot;ICANN&quot; domains)
+     * @return true if (re)initialization was successful
+     */
+    public boolean initialize(boolean excludePrivateDomains) {
+        return initialize(this.getClass().getResourceAsStream(ETLD_DATA), false);
+    }
+
+    /**
+     * (Re)initialize EffectiveTldFinder with custom public suffix list.
+     * 
+     * @param effectiveTldDataStream
+     *            content of public suffix list as input stream
+     * @return true if (re)initialization was successful
+     */
     public boolean initialize(InputStream effectiveTldDataStream) {
+        return initialize(effectiveTldDataStream, false);
+    }
+
+    /**
+     * (Re)initialize EffectiveTldFinder with custom public suffix list.
+     * 
+     * @param effectiveTldDataStream
+     *            content of public suffix list as input stream
+     * @param excludePrivateDomains
+     *            whether to exclude the public suffixes listed in the PRIVATE
+     *            domain section (opposed to &quot;ICANN&quot; domains)
+     * @return true if (re)initialization was successful
+     */
+    public boolean initialize(InputStream effectiveTldDataStream, boolean excludePrivateDomains) {
         domains = new HashMap<>();
+        boolean inPrivateDomainSection = false;
         try {
-            if (null == effectiveTldDataStream && null != this.getClass().getResource(ETLD_DATA)) {
-                effectiveTldDataStream = this.getClass().getResourceAsStream(ETLD_DATA);
-            }
             BufferedReader input = new BufferedReader(new InputStreamReader(effectiveTldDataStream, StandardCharsets.UTF_8));
             String line = null;
             while (null != (line = input.readLine())) {
-                if (line.length() == 0 || (line.length() > 1 && line.startsWith(COMMENT))) {
+                if (line.length() == 0) {
```

Comment:
Typically you want to trim the line before checking for empty lines or whether they start with a comment. Also `line.isEmpty()` is better here.

Reply:
agreed